app/modules/assets/models.py

```diff
@@ -255,6 +256,66 @@ def get_asset_sightings(self):
     def get_asset_group_sightings(self):
         return self.git_store.get_asset_group_sightings_for_asset(self)
 
+    @classmethod
+    def get_sage_sync_tags(cls):
+        return 'asset', 'image'
+
+    def sync_with_sage(self, ensure=False, force=False, bulk_sage_uuids=None, **kwargs):
+        from app.extensions.sage import from_sage_uuid
+
+        if force:
+            with db.session.begin(subtransactions=True):
+                self.content_guid = None
+                db.session.merge(self)
+            db.session.refresh(self)
+
+        if ensure:
+            if self.content_guid is not None:
+                if bulk_sage_uuids is not None:
+                    # Existence checks can be slow one-by-one, bulk checks are better
+                    if self.content_guid in bulk_sage_uuids.get('asset', {}):
+                        # We have found this Asset on Sage, simply return
+                        return
+                else:
+                    content_guid_str = str(self.content_guid)
+                    sage_rowids = current_app.sage.request_passthrough_result(
+                        'asset.exists', 'get', args=content_guid_str, target='sync'
+                    )
+                    if len(sage_rowids) == 1:
+                        if sage_rowids[0] is not None:
```

Comment:
second annoying style comment from me on the double ifs

Reply:
Consolidated